### PR TITLE
`azurerm_role_assignment` - supports "description", "condition", "condition_version"

### PR DESCRIPTION
**Pull request by njuCZ**
upgrade version for authorization from `2018-09-01-preview` to `2020-04-01-preview`

support property: "description", "condition", "condition_version"

docs about condition: https://review.docs.microsoft.com/en-us/azure/role-based-access-control/conditions-overview?branch=pr-en-us-119593

### CODE REVIEW

File: azurerm/internal/services/authorization/role_assignment_resource.go [hunk 1]
```diff
@@ -95,6 +95,28 @@ func resourceArmRoleAssignment() *schema.Resource {
...
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"condition": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
```
[the remark this answers is not in the document]
Comment (njuCZ) [reply azurerm/internal/services/authorization/role_assignment_resource.go:109]: I have asked the service team, they say it's a new syntax format. It's not easy to write the validation func

File: azurerm/internal/services/authorization/role_assignment_resource.go [hunk 2]
```diff
@@ -95,6 +95,28 @@ func resourceArmRoleAssignment() *schema.Resource {
...
+
+			"condition_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"1.0",
```
[the remark this answers is not in the document]
Comment (katbyte) [reply azurerm/internal/services/authorization/role_assignment_resource.go:119]: could we validate this?

**Comment by katbyte** [comment]
Could we also ensure all role definition tests pass to ensure the API upgrade isn't breaking anything? i see these two failing:
```
TestAccAzureRMRoleDefinition_update
TestAccAzureRMRoleDefinition_updateEmptyId
```

File: azurerm/internal/services/authorization/role_assignment_resource.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (katbyte) [reply azurerm/internal/services/authorization/role_assignment_resource.go:112]: should it at least be no emptry strings?

File: azurerm/internal/services/authorization/role_assignment_resource.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (njuCZ) [reply azurerm/internal/services/authorization/role_assignment_resource.go:122]: done

File: azurerm/internal/services/authorization/role_assignment_resource.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (njuCZ) [reply azurerm/internal/services/authorization/role_assignment_resource.go:114]: done

**Comment by njuCZ** [comment]
@katbyte it seems in the new api version, the update api needs more time to become eventually consistency. I have changed the `ContinuousTargetOccurence` from 6 to 10, all acctest in teamcity have passed.

**Review by katbyte — APPROVED** [review]
thanks @njuCZ - LGTM 👍

**Pull request merged**